docs/proposals/event-notification.md

```diff
@@ -0,0 +1,211 @@
+# ODO build and application status notification 
+
+## Summary
+
+With this proposal and [it's related issue](https://github.com/openshift/odo/issues/2550) we examine how to consume build/run event output from odo in order to allow external tools (such as IDEs) to determine the application/build status of an odo-managed application.
+
+In short, the proposal is:
+- New flag for push command, `odo push -o json`: Outputs JSON events that correspond to what devfile actions/commands are being executed by push.
+- New odo command `odo component status -o json`: Calls `supervisord ctl status` to check the running container processes, checks the container/pod status every X seconds (and/or a watch, for Kubernetes case), and sends an HTTP/S request to the application URL. The result of those is output as JSON to be used by external tools to determine if the application is running.
```

Comment:
Hmm. Haven't heard of this before. @kadel thoughts? Have there been other discussions elsewhere?

Reply:
This proposal was discussed in this PR https://github.com/openshift/odo/pull/3177
The PR was by mistake opened against `parser-v2` branch, so this is why it shows up here.